feat(style): add TailwindCSS colors

joshka opened the pull request:
Enable the `tailwind` feature flag to use the colors from the Tailwind
CSS framework. The colors are available in the `style::tailwind` module.

See https://tailwindcss.com/docs/customizing-colors for more information
on Tailwind.

```rust
use ratatui::style::tailwind::SLATE;
Line::styled("Hello", SLATE.c500);
```

kdheepak commented:
Should we make the user facing syntax like this:

```
use ratatui::style::palette::tailwind::Blue;
```

? It is more clear that anything under `palette` is associated with color palettes.

joshka commented:
![image](https://github.com/ratatui-org/ratatui/assets/381361/23197fd2-6a3f-46be-8b0c-e14daa616c8d)

Valentin271 commented:
I'd go for `ratatui::style::theme` but I'm fine with `ratatui::theme` too.

joshka commented:
> I'd go for `ratatui::style::theme` but I'm fine with `ratatui::theme` too.

What about `ratatui::bikeshed::`? ;P

kdheepak commented:
I like `ratatui::style::theme` too.

kdheepak commented:
Thinking about it more, I _think_ `theme` isn't quite the right word for this. 

In other apps, vim, vscode etc, a `theme` is a mapping of UI elements to a color. I think `palette` is the right word for what this PR introduces. It's what the tailwindcss documentation uses on that page to describe the colors as well.

I can imagine a future feature where we add a `ratatui::theme::Theme` struct that maps UI elements to colors, and are applied universally to any widget you pick.

joshka commented:
I went with `ratatui::style::palette::tailwind` for this. Rebased, squashed and ready to merge. Once this is merged I'll rebase the material branch on it.
